pandas/core/strings.py

```diff
@@ -372,21 +377,33 @@ def str_replace(arr, pat, repl, n=-1, case=True, flags=0):
     0    tWO
     1    bAR
     dtype: object
+
+    When `pat` is a compiled regex, all flags should be included in the
+    compiled regex. Use of `case` and `flags` with a compiled regex will
+    raise an error.
+
+    >>> regex_pat = re.compile(r'FUZ', flags=re.IGNORECASE)
+    >>> pd.Series(['foo', 'fuz', np.nan]).str.replace(regex_pat, 'bar')
+    0    foo
+    1    bar
+    2    NaN
+    dtype: object
     """
 
     # Check whether repl is valid (GH 13438, GH 15055)
     if not (is_string_like(repl) or callable(repl)):
         raise TypeError("repl must be a string or callable")
-    use_re = not case or len(pat) > 1 or flags or callable(repl)
+    # Check whether pat is a compiled regex or should be compiled
+    is_compiled_re = isinstance(pat, type(re.compile("")))
+    use_re = (is_compiled_re or not case or
+              len(pat) > 1 or flags or callable(repl))
```

Comment:
I would have an immediate check after ``is_compiled_re`` that sees if ``case/flags`` are passed (and raise)